DataDog/chaos-controller
# Write spec hash in resource status to detect resource changes

### What does this PR do?

On disruption resource creation, it computes the spec hash and stores it in the resource status. If the hash changes later, it means that the spec has been changed and an event is registered in the resource to warn the user about its immutability.

### Motivation

Warn the user about the disruption resource immutability when a change is made on the spec.

### Testing Guidelines

* create a disruption resource
* describe it
  * the status should contain the hash
  * no events about the immutability should be registered
* edit the resource by changing something in the spec
* describe it again
  * the hash in the status should remain the same
  * an event about the immutability of the resource should be registered
